src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.exception.EmptyTodoDescriptionException;
+import duke.exception.UnknownCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * deals with parsing the user command
+ */
+public class Parser {
+
+    /**
+     * @param fullCommand the user command
+     * @return a duke.command.Command object which encapsulates the information of a parsed user command
+     * @throws DukeException exception when there is an parsing error
+     */
```

Comment:
Maybe a short summary of the method could be written as the first sentence for this method header.